eng/scripts/FinishDumpCollectionForHangingBuilds.ps1

```diff
@@ -0,0 +1,74 @@
+param(
+  [Parameter(Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string]
+  $ProcDumpDumpFolderPath
+)
+
+Write-Output "Finishing dump collection for hanging builds.";
+
+$repoRoot = Resolve-Path "$PSScriptRoot\..\..";
+$ProcDumpDumpFolderPath = Join-Path $repoRoot $ProcDumpDumpFolderPath;
+
+$sentinelFile = Join-Path $ProcDumpDumpFolderPath "dump-sentinel.txt";
+if ((-not (Test-Path $sentinelFile))) {
+  Write-Output "No sentinel file available in '$sentinelFile'.";
+  return;
+}
+
+Get-Process "procdump" -ErrorAction SilentlyContinue | ForEach-Object { Write-Output "ProcDump with PID $($_.Id) is still running."; };
+
+$capturedDumps = Get-ChildItem $ProcDumpDumpFolderPath -Filter *.dmp;
+$capturedDumps | ForEach-Object { Write-Output "Found captured dump $_"; };
+
+$JobName = (Get-Content $sentinelFile);
+
+if ($JobName.Count -ne 1) {
+  if ($JobName.Count -eq 0) {
+    Write-Warning "No job name found. This is likely an error.";
+    return;
+  }
+  else {
+    Write-Output "Multiple job names found '$JobName'.";
+    return;
+  }
+}
+
+$dumpCollectionJob = Get-Job -Name $JobName -ErrorAction SilentlyContinue;
+$registeredJob = Get-ScheduledJob -Name $JobName -ErrorAction SilentlyContinue;
+
+if ($null -eq $dumpCollectionJob) {
+  Write-Output "No job found for '$JobName'. It either didn't run or there is an issue with the job definition.";
+
+  if ($null -eq $registeredJob) {
+    Write-Warning "Couldn't find a scheduled job '$JobName'.";
+  }
+  return;
+}
+
+Write-Host "Listing existing jobs";
+Get-Job
+Write-Host "Listing existing scheduled jobs";
+Get-ScheduledJob
```

Comment:
Why is this output useful?

Reply:
Debug purposes. Registering the Job is trickier than it seems.

Reply:
Now you've figured it out, listing all jobs on the agent will just be confusing. Please remove or at least limit the output to the `CaptureDumps*` jobs.

Reply:
Same reason as for the other comment. This information was useful in troubleshooting issues, its useful to keep it here for troubleshooting future issues.

I don't see how this is confusing, the output clearly states what the information being displayed is and if you don't think you need it, you can simply skip over it.